### PR TITLE
Update lora merging tutorials and update LoRA evaluation info

### LINKED ISSUE #494: Needing a LoRA-weight merging script for real-world use
I mentioned before that we could really use a script that merges LoRA weights with the base model to get an updated checkpoint model (tagging you @carmocca @JustinGoheen since I think we discussed this before, and I wanted to raise an additional point in favor of such a script).

The script's usage could be 

```bash
python finetune/lora.py \ # as usual
--checkpoint_dir base_model
--out_dir lora_weights

python scripts/merge_lora_weights.py \ # new
--checkpoint_dir base_model
--lora_weights lora_weights
--out_dir new_checkpoint
```

## Overcoming finetuning limitations

As far as I know, it's currently not possible to finetune a model on multiple datasets due to the following limitation:

```bash
python finetune/lora.py \
--checkpoint_dir basemodel \
--data dir dolly \
--out_dir lora_weights_1

python finetune/lora.py \
--checkpoint_dir ??? \
--data dir lima \
--out_dir lora_weights_2
```

Right now, finetuning on multiple datasets would only be possible with adapters, as far as I know?

```bash
python finetune/adapter.py \
--checkpoint_dir basemodel \
--data dir dolly \
--out_dir finetuned_model_stage_1

python finetune/adapter.py \
--checkpoint_dir finetuned_model_stage_1 \
--data dir lima \
--out_dir finetuned_model_stage_2
```

## Simplifying the repository

Also, if we had such a script, we wouldn't need to maintain separate evaluation scripts like [lm_eval_harness.py](https://github.com/Lightning-AI/lit-gpt/blob/main/eval/lm_eval_harness.py) and [lm_eval_harness_lora.py](https://github.com/Lightning-AI/lit-gpt/blob/main/eval/lm_eval_harness_lora.py)

### PR DESCRIPTION
**Pull request by rasbt**
The evaluation script for LoRA had a few problems, but good news is that we now have the LoRA merge script from #495 so that we don't need a dedicated Evaluation Harness script for LoRA anymore, so I removed it. I updated the documentation accordingly.

Closes #494

### CODE REVIEW

File: tutorials/finetune_lora.md [hunk 1]
```diff
@@ -160,11 +160,16 @@ Then, we can merge the LoRA weights with the checkpoint model using the `merge_l
...
+  --lora_path "out/lora_weights/stablelm-base-alpha-3b/lit_model_lora_finetuned.pth" \
   --out_dir "out/lora_merged/stablelm-base-alpha-3b/"
 ```
 
-After merging, we can use the `base.py` file for inference using the new checkpoint file:
+After merging, we can use the `base.py` file for inference using the new checkpoint file. Note that if your new checkpoint directory is different from the original checkpoint directory, we also have to copy over the `*.json` files which are required for the configuration and tokenizer information:
+
+```bash
+scp checkpoints/stabilityai/stablelm-base-alpha-3b/*.json \
+out/lora_merged/stablelm-base-alpha-3b/
```
Comment (carmocca) [inline tutorials/finetune_lora.md:171]: nit: cp would work too

```suggestion
cp checkpoints/stabilityai/stablelm-base-alpha-3b/*.json out/lora_merged/stablelm-base-alpha-3b/
```

File: tutorials/neurips_challenge_quickstart.md [hunk 2]
```diff
@@ -193,9 +193,36 @@ python eval/lm_eval_harness.py \
...
+```
+
+```bash
+scp checkpoints/stabilityai/stablelm-base-alpha-3b/*.json \
+out/lora_merged/stablelm-base-alpha-3b/
+```
 
+For more information on LoRA weight merging, please see the
+[Merging LoRA Weights](https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/finetune_lora.md#merging-lora-weights)
+section of the LoRA finetuning documentation.
```
Comment (carmocca) [inline tutorials/neurips_challenge_quickstart.md:213]: Should we just link to the lora weight merging section?

```suggestion
To evaluate a LoRA-finetuned model, you need to first merge the LoRA weights with the base model to create a new checkpoint file. Please follow the
[Merging LoRA Weights](https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/finetune_lora.md#merging-lora-weights)
section of the LoRA finetuning documentation.
```

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
Also, shouldn't you remove the `_lora` file here?

**Comment by rasbt** [comment]
> Also, shouldn't you remove the `_lora` file here?

Arg, my bad, I used `git add tutorials/` 

> Should we just link to the lora weight merging section?

Yes, that was what I was originally thinking, but in my experience, it's best to have this starter kit self-contained. Something that someone can follow without having to necessarily go look elsewhere. I.e., making it as easy as possible for newcomers, like following a recipe.

> nit: cp would work too

I don't have a preference since this is locally it doesn't matter I guess. I changed it.

**Pull request merged**